src/main/java/org/biscuitsec/biscuit/token/builder/parser/Parser.java

```
@@ -127,6 +148,42 @@ public static Either<Map<Integer, List<Error>>, Block> datalog(long index, Strin
       return Either.left(errors);
     }
 
+    return Either.right(new Tuple5<>(facts, rules, checks, scopes, policies));
+  }
+
+  /**
+   * Takes a datalog string with <code>\n</code> as datalog line separator. It
+   * tries to parse each
+   * line using fact, rule, check and scope sequentially.
+   *
+   * <p>
+   * If one succeeds it returns Right(Block) else it returns a Map[lineNumber,
+   * List[Error]]
+   *
+   * @param index block index
+   * @param s     datalog string to parse
+   * @return Either<Map<Integer, List<Error>>, Block>
+   */
+  public static Either<Map<Integer, List<Error>>, Block> datalog(long index, String s) {
```

a non-empty policy list should be an error here imo. afaik, policies are silently ignored here